mirrord/agent/src/entrypoint.rs

```diff
@@ -488,10 +489,19 @@ impl ClientConnectionHandler {
 }
 
 /// Initializes the agent's [`State`], channels, threads, and runs [`ClientConnectionHandler`]s.
-#[tracing::instrument(level = "trace", ret)]
-async fn start_agent(args: Args) -> Result<()> {
+#[tracing::instrument(level = Level::TRACE, ret, err)]
+async fn start_agent(args: Args) -> AgentResult<()> {
     trace!("start_agent -> Starting agent with args: {args:?}");
 
+    if let Some(metrics_address) = args.metrics.as_ref() {
+        let address = metrics_address.parse()?;
```

Comment:
You can store the address as `SocketAddr` in both agent config and agent CLI params, no need to parse here and add a very generic `AgentError` variant :eyes: